package/MDAnalysis/lib/distances.py

```diff
@@ -208,6 +208,23 @@ def _check_lengths_match(*arrays):
         raise ValueError("Input arrays must all be same shape"
                          "Got {0}".format([a.shape for a in arrays]))
 
+
+def distance(a, b, box=None):
+    """calc distances between two points. Uses algorithm from Tuckerman
+
+    """
+    if box is None:
+        return np.linalg.norm(a - b)
+
+    box = triclinic_vectors(box)
+    ibox = np.linalg.inv(box)
```

Comment:
Yes with cython this can be speed up. But I have to rewrite to use less variables to get the full speedup and not keep calling python functions.

```python
from numba import jit
import numpy as np

@jit
def jit_ibox(box):
    inv = np.zeros(box.shape)
    inv[0, 0] = 1 / box[0, 0]
    inv[1, 0] = -box[1,0] / (box[0,0]*box[1,1])
    inv[1, 1] = 1 / box[1,1]
    inv[2, 0] = (box[1,0]*box[2,1] - box[1,1]*box[2,0])/(box[0,0]*box[1,1]*box[2,2])
    inv[2, 1] = -box[2,1]/(box[1,1]*box[2,2])
    inv[2, 2] = 1 / box[2,2]
    return inv
```

*note*: I tested this with numba because it has similar performance but much shorter test cycles

```
%timeit jit_ibox(box)
656 ns ± 18.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```